cadl-extension/src/code-model-builder.ts

```diff
@@ -679,6 +701,55 @@ export class CodeModelBuilder {
           }
         }
         request.signatureParameters = request.parameters;
+
+        if (request.signatureParameters.length > 6) {
```

Comment:
Currently, just hardcode a "6". And this at present only apply when flatten happens.